cmd/teleporter-cli/root.go

```diff
@@ -0,0 +1,81 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package main
+
+import (
+	"os"
+
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/subnet-evm/accounts/abi"
+	teleportermessenger "github.com/ava-labs/teleporter/abi-bindings/go/Teleporter/TeleporterMessenger"
+	"github.com/spf13/cobra"
+)
+
+var (
+	logger        logging.Logger
+	teleporterABI *abi.ABI
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "teleporter-cli",
+	Short: "A CLI that integrates with the Teleporter protocol",
+	Long: `A CLI that integrates with the Teleporter protocol, and allows you
+to debug Teleporter on chain activity. The CLI can help decode
+Teleporter and Warp events, as well as parsing Teleporter messages.`,
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	err := rootCmd.Execute()
+	if err != nil {
+		os.Exit(1)
+	}
+}
+
+func init() {
+	rootCmd.CompletionOptions.DisableDefaultCmd = true
+	logLevelArg := rootCmd.PersistentFlags().StringP("log", "l", "", "Log level i.e. debug, info...")
+	rootCmd.PersistentPreRunE = func(cmd *cobra.Command, args []string) error {
+		return rootPreRunE(logLevelArg)
+	}
+}
+
+func rootPreRunE(logLevelArg *string) error {
```

Comment:
Possibly dumb question, but what's the `E` suffix denote here and in `callPersistentPreRunE`, etc?

Reply:
In `cobra` cli there's commonly functions with an extra `E` suffix as duplicates, i.e. `PersistentPreRun` vs `PersistentPreRunE`. The one's with E just mean return an error